vars/buildPlugin.groovy

```diff
@@ -102,6 +106,35 @@ def call(Map params = [:]) {
                     '-Dcheckstyle.failOnViolation=false',
                     '-Dcheckstyle.failsOnError=false',
                   ]
+                  if (artifactCachingProxyEnabled) {
+                    withCredentials([
+                      usernamePassword(credentialsId: 'artifact-caching-proxy-credentials', usernameVariable: 'ARTIFACT_CACHING_PROXY_USERNAME', passwordVariable: 'ARTIFACT_CACHING_PROXY_PASSWORD')
+                    ]) {
+                      final String defaultProxyProvider = 'azure'
+                      def availableProxyProviders = ['azure', 'do', 'aws']
+                      String requestedProvider = env.ARTIFACT_CACHING_PROXY_PROVIDER
+                      String mavenSettings = libraryResource 'artifact-caching-proxy-settings.xml'
+
+                      // As azure VM agents don't have this env var, setting a default provider if none is specified or if the provider isn't available
+                      if (requestedProvider == null || requestedProvider == '' || !availableProxyProviders.contains(requestedProvider)) {
+                        requestedProvider = defaultProxyProvider
+                        echo "INFO: no valid artifact caching proxy provider specified, set to '$defaultProxyProvider' by default."
+                      }
+
+                      echo "Setting up Maven to use artifact caching proxy from '${requestedProvider}' provider..."
+
+                      settingsFile = "${pwd tmp: true}/mavenSettings.xml"
```

Comment:
```suggestion
                      settingsFile = "$WORKSPACE_TMP/mavenSettings.xml"
```
simpler

Reply:
I've reproduced [what was above](https://github.com/jenkins-infra/pipeline-library/pull/499/files#diff-db2794773b738dbb5566313f115cdf3ff1f45e13e57cbc37aa1bd181819ed5fbR99).
I agree it's simpler, will change it depending on the direction of this PR.